app/position/positionprovidersmodel.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef POSITIONPROVIDERSMODEL_H
+#define POSITIONPROVIDERSMODEL_H
+
+#include <QObject>
+#include <qglobal.h>
+#include <QAbstractListModel>
+
+class AppSettings;
+
+struct PositionProvider
```

Comment:
I personally do not like structs :). Alternative to this is c++ class where all members are not modifiable/final (?) and you have only gettters. Anyways, this is good too, matter of taste